### PR TITLE
define copy constructor if assignment operator is defined

### PR DESCRIPTION
**Pull request by rodolforg**
Reason: c++ Rule of 5 (by using copy-swap idiom)

### CODE REVIEW

**Comment by rodolforg** [comment]
Oops, I messed up here

**Comment by rodolforg** [comment]
And now I fixed :P

File: synfig-core/src/synfig/value.cpp [hunk 1]
```diff
@@ -210,16 +218,15 @@ ValueBase::clear()
...
 	type=&type_nil;
 }
 
-
 Type&
 ValueBase::ident_type(const String &str)
 {
 	Type *type = Type::try_get_type_by_name(str);
-	return type == NULL ? type_nil : *type;
+	return !type ? type_nil : *type;
```
Comment (ice0) [inline synfig-core/src/synfig/value.cpp]: Maybe better to invert the condition now? Like:
```
return type ? *type : type_nil;
```

Comment (rodolforg) [reply synfig-core/src/synfig/value.cpp]: fixed and squashed

**Pull request merged**